Automation/create_vm_script.ps1

```diff
@@ -0,0 +1,98 @@
+#This script creates a Ubuntu VM on Azure.
+#The script logs user into Azure using Service Principal. To log in,
+#Tenant ID, App ID, Secret, and Subscription ID are required.
+#User must provide Azure Resource Group Name, VM Name, Admin Username/Password
+
+param (
+    [string] [Parameter(Mandatory=$true)] $tenantId, #Tenant ID used to login to Azure with Service Principal
+    [string] [Parameter(Mandatory=$true)] $applicationId, #App ID used to login to Azure with Service Principal
+    [string] [Parameter(Mandatory=$true)] $secret, #Azure Secret used to login to Azure with Service Principal
+    [string] [Parameter(Mandatory=$true)] $subscriptionId, #Subscr. ID used to login to Azure with Service Principal
+    [string] [Parameter(Mandatory)] $rgName, #Resource Group Name required
+    [string] [Parameter(Mandatory)] $vmName, #VM Name required
+    [string] [Parameter(Mandatory)] $adminUsername, #Admin Username required
+    [String] [Parameter(Mandatory)] $adminPassword, #Admin Password required
+    [string] $location='West US2', #Default value for RG/VM Location
+    [string] $vNetName='myVnet',  #Default value for Virtual Network Name
```

Comment:
for default values maybe you could concatenate the deployment name (would need to add as a parameter) with "-nsg" and "-vnet", etc

Reply:
great idea